plugins/inputs/t128_tank/t128_tank.go

```diff
@@ -118,20 +126,40 @@ func (plugin *T128Tank) Stop() {
 	plugin.mainWG.Wait()
 }
 
-func extractTopicType(message string, pattern string) string {
-	regex := regexp.MustCompile(pattern)
-	match := regex.FindStringSubmatch(message)
-	if len(match) > 1 {
-		return match[1]
-	}
-	return ""
-}
-
 func (plugin *T128Tank) checkConfig() error {
 	if plugin.Topic == "" {
 		return fmt.Errorf("topic is a required configuration field")
 	}
 
+	if plugin.From != "" {
+		err := validateFrom(plugin.From)
+		if err != nil {
+			return fmt.Errorf("%s", err)
+		}
+	}
+
+	if strings.ToLower(plugin.From) == "end" {
+		plugin.defaultStartingIndex = EndIndex
+	} else if strings.ToLower(plugin.From) == "start" {
+		plugin.defaultStartingIndex = StartIndex
+	} else {
+		if plugin.IndexFile == "" {
+			plugin.defaultStartingIndex = EndIndex
+		} else {
+			plugin.defaultStartingIndex = StartIndex
+		}
+	}
+	return nil
+}
+
+func validateFrom(from string) error {
+	validFromValues := map[string]bool{
```

Comment:
There are so few values here, it is probably best just to compare against the two strings

```
fromLower := strings.ToLower(from)

if fromLower != "start" && fromLower == "end {
    return errors.New("...")

return nil
```

Reply:
There was a typo in the snipped above. Fixed now